compiler/rustc_hir_analysis/src/coherence/inherent_impls_overlap.rs

```diff
@@ -205,11 +210,6 @@ impl<'tcx> InherentOverlapChecker<'tcx> {
             // This is advantageous to running the algorithm over the
             // entire graph when there are many connected regions.
 
-            rustc_index::newtype_index! {
-                #[orderable]
-                pub struct RegionId {}
-            }
```

Comment:
Why does this trigger the lint? Aren't all the impls on `RegionId` that is defined locally here?

Reply:
This is because of some `From` impls in the generated code https://github.com/rust-lang/rust/blob/master/compiler/rustc_index_macros/src/newtype.rs#L288C13-L314C14